tests/js/server/aql/aql-graph-traversal-generic-noncluster.js

```diff
@@ -25,18 +25,31 @@
 /// @author Heiko Kernbach
 ////////////////////////////////////////////////////////////////////////////////
 
-const {protoGraphs} = require('@arangodb/testutils/aql-graph-traversal-generic-graphs.js');
+const {protoGraphs, TestVariants} = require('@arangodb/testutils/aql-graph-traversal-generic-graphs.js');
 const {testsByGraph, metaTests} = require('@arangodb/testutils/aql-graph-traversal-generic-tests.js');
 const console = require('console');
 const jsunity = require("jsunity");
 const _ = require("lodash");
+const isEnterprise = require("internal").isEnterprise();
 
 function graphTraversalGenericGeneralGraphStandaloneSuite() {
   let testGraphs = _.fromPairs(_.keys(protoGraphs).map(x => [x, {}]));
   _.each(protoGraphs, function (protoGraph) {
     _.each(protoGraph.prepareSingleServerGraph(), function (testGraph) {
-      testGraphs[protoGraph.name()][testGraph.name()] = testGraph;
+      testGraphs[protoGraph.name()][`${testGraph.name()}_SingleServerGeneralGraph`] = testGraph;
     });
+
+    if (isEnterprise) {
+      _.each(protoGraph.prepareSmartGraphs(TestVariants.SmartGraphSingleServer), function (testGraph) {
```

Comment:
I understand what you are trying to do here, but can we not simply join `generic-cluster` and `generic-noncluster` versions of this file?
This way we make sure the exact same test is executed on SingleServer and Cluster.
This seems to be triggered with the following code in connection with the modified helper library.

Also in general I would prefer to have an enterprise version of this file instead of the `isEnterprise` if here.
Happy to jump on a call to explain what I am after, to make this setup easier.

(I think what you test here is correct nevertheless)

Reply:
Merged cluster and non-cluster variant :)